### PR TITLE
User-editable information

### PR DESCRIPTION
**Pull request by filiptypjeu**
Adds support for users to edit their own information on their profile. Also enhances the public view of profiles in general, and includes minor fixes all-around. This resolves issues #70, #122, #194 and #205.

### CODE REVIEW

File: teknologr/katalogen/templates/profile_information.html [hunk 1]
```diff
@@ -0,0 +1,208 @@
...
+<div
+  id="member-information"
+  class="col-12 col-lg-7"
+  hx-target="#member-information"
+  hx-swap="outerHTML show:bottom transition:true"
+>
+  <table class="table table-striped table-borderless">
+    <tbody>
+      <tr>
+        <th>Födelseår</th>
```
Comment (backjonas) [inline teknologr/katalogen/templates/profile_information.html:12]: I would be very careful when increasing the amount of personal information we show to other people and would prefer to keep all birth time related info hidden from other people as it was before, unless you've discussed it already and agree that it should be shown.

File: teknologr/katalogen/views.py [hunk 2]
```diff
@@ -65,17 +74,63 @@ def profile(request, member_id):
...
+    own_profile_or_403(request, member)
+
+    if request.method == 'POST':
+        form = EditProfileForm(request.POST, instance=member)
+        if form.is_valid():
+            try:
+                form.save()
+                form = None
+            except LDAPError:
+                form.add_error('email', 'E-postadressen kunde tyvärr inte uppdateras, vänligen kontakta en administrator')
```
Comment (backjonas) [inline teknologr/katalogen/views.py]:
```suggestion
                form.add_error('email', 'E-postadressen kunde tyvärr inte uppdateras, vänligen kontakta en administratör')
```
🤓

File: teknologr/katalogen/templates/profile_information.html [hunk 3]
```diff
@@ -0,0 +1,208 @@
...
+        </td>
+      </tr>
+      <tr>
+        <th>Ordinarie medlem</th>
+        <td class="text-right">
+          {{ person.omembertime_text }}
+        </td>
+      </tr>
+      <tr>
+        <th>Ständig äldre medlem</th>
```
Comment (backjonas) [inline teknologr/katalogen/templates/profile_information.html:40]: Only show if stälm?

Same for other fields here

File: teknologr/katalogen/templates/profile_information.html [hunk 4]
```diff
@@ -0,0 +1,208 @@
...
+          Utexamineringsår
+          {{ form.graduated.errors }}
+          {{ form.graduated_year.errors }}
+        </th>
+        <td class="text-right">
+          {% firstof form.graduated_year person.graduated_year %}
+        </td>
+      </tr>
+      <tr>
+        <th>
```
Comment (backjonas) [inline teknologr/katalogen/templates/profile_information.html:127]: Not sure at all if this is needed or even helpful, but should the subscribe & allow info fields have the same additional info tooltip as in the registration form?

**Review by backjonas — APPROVED** [review]
Looks good, great work!

File: teknologr/katalogen/templates/profile_information.html [hunk 5]
```diff
@@ -0,0 +1,208 @@
...
+          Prenumerera på Studentbladet
+          {{ form.allow_studentbladet.errors }}
+        </th>
+        <td class="text-right">
+          {% firstof form.allow_studentbladet person.allow_studentbladet_text %}
+        </td>
+      </tr>
+      <tr>
+        <th>
+          Visa kontaktinformation
```
Comment (djstockma) [inline teknologr/katalogen/templates/profile_information.html]: Maybe something like "tillåt visning av kontaktinformation"

File: teknologr/members/templates/applicant.html [hunk 6]
```diff
@@ -115,18 +100,10 @@ <h5 class="my-auto">
...
-                autocomplete="off"
-                {% if form.allow_studentbladet.value %}checked{% endif %}
-              >
-              <label class="form-check-label pl-2" for="{{ form.allow_studentbladet.id_for_label }}">
-                Jag vill få Studentbladet hemskickad och tillåter att<br />
-                mina uppgifter utlämnas till Studentbladets redaktion
+              {{ form.allow_publish_info.errors }}
+              <label class="form-check-label pl-2" for="{{ form.allow_publish_info.id_for_label }}">
+                {{ form.allow_publish_info }}
+                Jag tillåter att mina uppgifter utlämnas till Katalogen
```
Comment (djstockma) [inline teknologr/members/templates/applicant.html]: Should this be a bit more descriptive and/or explain what this means in practice? Also, should this same explanation be given in the edit view (for atleast this consent checkbox)

**Review by djstockma — APPROVED** [review]
Looks very good and solves multiple actual problems!

File: teknologr/katalogen/views.py (hunk 2 above)
Comment (filiptypjeu) [reply teknologr/katalogen/views.py]: TIL

File: teknologr/katalogen/templates/profile_information.html (hunk 4 above)
Comment (filiptypjeu) [reply teknologr/katalogen/templates/profile_information.html:127]: Yes, it probably should. But I'm too lazy to add that now. I'll create an issue and leave that for later.

File: teknologr/katalogen/templates/profile_information.html (hunk 5 above)
Comment (filiptypjeu) [reply teknologr/katalogen/templates/profile_information.html]: Yes, that is perhaps better

File: teknologr/members/templates/applicant.html (hunk 6 above)
Comment (filiptypjeu) [reply teknologr/members/templates/applicant.html]: This is the applicant (Ansökande) page on the admin site, so I'd say it's not too important to be overly descriptive. But could of course be consistent.

File: teknologr/katalogen/templates/profile_information.html (hunk 3 above)
Comment (filiptypjeu) [reply teknologr/katalogen/templates/profile_information.html:40]: The fields are left empty on purpose. Whether or not this is correct can of course be discussed, but leaving it as is for the time being.

File: teknologr/katalogen/templates/profile_information.html (hunk 1 above)
Comment (filiptypjeu) [reply teknologr/katalogen/templates/profile_information.html:12]: Birth year has historically been displayed in (the physical) Katalogen, which is why I opted to show it here as well. I will remove it for the time being.

**Pull request merged**